addons/zylann.hterrain/hterrain_data.gd

```diff
@@ -1426,7 +1426,72 @@ func _import_heightmap(fpath: String, min_y: int, max_y: int, big_endian: bool)
 				f.get_16()
 
 		im.unlock()
+	elif ext == "xyz":
+		_logger.debug("detected xyz file")
+		var f := File.new()
+		var err := f.open(fpath, File.READ)
+		if err != OK:
+			return false
+
+		var line = f.get_line()
+		# f.get_8() get byte to string if \r then win ending \r\n (2) else linux ending \n (1)
+		var file_len = f.get_len()
+		var file_res = Util.integer_square_root(file_len / (line.length() + 2)) # + 2 ist /r/n should be dynamic
+		var res := get_adjusted_map_size(file_res, file_res)
```

Comment:
you are right. get_8() used to be there, before I tried to do it with get_line so line 1437 is obsolete.

also yes it is bad to get a line.length for that solution, for some reason the line length in my file are always normalised to be 27characters long.

the better way would be to open the file at the end and get the last line, but I don't know of a way to do this in GDScript. The other way would be to do a "get_line" until I'm at the end to seek the file, but this would cause quite a big memory overhead and would also take a while as the example file alone is 110mb large.